src/transformers/models/t5/modeling_tf_t5.py

```diff
@@ -1191,24 +1189,24 @@ def __init__(self, config, *inputs, **kwargs):
         if not config.tie_word_embeddings:
             self.lm_head = tf.keras.layers.Dense(config.vocab_size, use_bias=False, name="lm_head")
 
-    def get_input_embeddings(self):
-        return self.shared
-
     def get_output_embeddings(self):
         if self.config.tie_word_embeddings:
-            return self.shared
+            return self.get_input_embeddings()
         else:
-            return self.lm_head
-
-    def set_input_embeddings(self, new_embeddings):
-        self.shared.weight = new_embeddings
-        # retrieve correct absolute scope for embed token wrapper
-        with tf.compat.v1.variable_scope("shared") as shared_abs_scope_name:
-            pass
-        # Wraps layer to avoid problems with weight restoring and ensuring we're in the correct TF scope.
-        embed_tokens = TFWrappedEmbeddings(self.shared, abs_scope_name=shared_abs_scope_name)
-        self.encoder.set_embed_tokens(embed_tokens)
-        self.decoder.set_embed_tokens(embed_tokens)
+            # in a dense layer the kernel has a shape (last_dim, units), for us (dim, num_tokens)
+            # value has a shape (num_tokens, dim) then needs to be transposed
+            return tf.transpose(self.lm_head.kernel)
+
+    def set_output_embeddings(self, value):
+        if value is not None:
```

Comment:
why would the value be `None`? I'd delete this if

Reply:
This is a public method, we never know how this method will be used, so to prevent at least this misusage I have added this condition.

Reply:
It's better to delete it still. If a user calls 

```python

t5.set_output_embeddings(self, word_embeddings)
```

and `word_embeddings` is None for whatever reason this should raise an error IMO not just silently ignore it. If we ignore it the user might have a much harder time finding the error.

Reply:
Ok, let's remove this condition then.